autrainer/training/training.py

```diff
@@ -800,34 +800,20 @@ def _disaggregated_evaluation(self, df, groundtruth, stratify):
         TODO: Need to define schema in the docs.
 
         """
-        df = df.set_index(groundtruth.index)
```

Comment:
Ah, good point. This is not something I had in mind, I thought `_disaggregated_evaluations` is only called for dev/test (i.e. without shuffling). 

However, I am not using `df` at all here, but rather get the indices from `groundtruth`:

https://github.com/autrainer/autrainer/blob/30d76ecf5e6071ec864fa1b54266663ec26290dd/autrainer/training/training.py#L824-L828

This is still wrong because `idx` corresponds to the original index. I can change this to:

```python
idx = groundtruth.loc[groundtruth[s] == v].index
                    # Map groundtruth indices to tracker indices
                    # This accounts for random shuffling
                    indices = [
                        i
                        for i, x in enumerate(tracker.indices)
                        if x in idx
                    ]
                    results[metric.name][v] = metric(
                        tracker.targets[indices],
                        tracker.predictions[indices],
                    )
``` 
Do you see a faster implementation?

See 16f06dc

P.S. This also needs a test though..

Reply:
I moved `disaggregated_evaluation` to `autrainer.training.utils` and so was able to test it in `test_metrics`, I've also changed the interface a bit

Reply:
Okay yeah i think this should be fine and the tests (last one for disaggregated evaluation) also cover shuffled indices right?

Reply:
Not all tests cover disaggregated indices, only the last one